tree/ntupleutil/v7/src/RNTupleImporter.cxx

```diff
@@ -173,6 +173,8 @@ ROOT::Experimental::RResult<void> ROOT::Experimental::RNTupleImporter::PrepareSc
          return R__FAIL("unsupported: count leaf arrays in leaf list, branch " + std::string(b->GetName()));
 
       // Only plain leafs with type identifies 'C' are C strings. Otherwise, they are char arrays.
+      // We use GetLeafCounter instead of GetLeafCount and GetLenStatic because the latter don't distinguish between
+      // char arrays and C strings.
```

Comment:
Lightly scanning the code of `GetLeafCounter` I don't exactly see where it 'distinguishes` between `char arrays` and `C strings`.  Can you expand the explanation a bit more?

Reply:
I think it's due to the different titles. C string read `myString/C` and char arrays `c[4]/C`.